### PR TITLE
AP_Periph: fixed moving baseline yaw for single CAN port GPS peripherals

### PR DESCRIPTION
**Pull request by tridge**
The AP_GPS autoconfig code for moving baseline GPS assumes that the GPS_MB_CAN_PORT parameter is available on the node, and if not then it shows GPS status as 1 and never gets lock
This works around that. We could instead try to fix AP_GPS_UAVCAN.cpp to handle the error from a missing param

### CODE REVIEW

File: Tools/AP_Periph/Parameters.h [hunk 1]
```diff
@@ -95,7 +95,7 @@ class Parameters {
 
 #ifdef HAL_PERIPH_ENABLE_GPS
     AP_Int8 gps_port;
-#if HAL_NUM_CAN_IFACES >= 2
+#if HAL_NUM_CAN_IFACES >= 2 || GPS_MOVING_BASELINE
```
Comment (bugobliterator) [inline Tools/AP_Periph/Parameters.h:98]: I think just having GPS_MOVING_BASELINE is enough.

File: Tools/AP_Periph/Parameters.cpp [hunk 2]
```diff
@@ -173,7 +173,7 @@ const AP_Param::Info AP_Periph_FW::var_info[] = {
     // @RebootRequired: True
     GSCALAR(gps_port, "GPS_PORT", HAL_PERIPH_GPS_PORT_DEFAULT),
 
-#if HAL_NUM_CAN_IFACES >= 2
+#if HAL_NUM_CAN_IFACES >= 2 || GPS_MOVING_BASELINE
```
Comment (bugobliterator) [inline Tools/AP_Periph/Parameters.cpp:176]: Just checking for GPS_MOVING_BASELINE is good enough

**Review by bugobliterator — DISMISSED** [review]

**Comment by rmackay9** [comment]
Awesome, thanks!

I've put a post here so we can get the beta testers to test it. https://discuss.ardupilot.org/t/copter-4-2-0-beta1-available-for-beta-testing/82460/15

File: Tools/AP_Periph/Parameters.h (hunk 1 above)
Comment (tridge) [reply Tools/AP_Periph/Parameters.h:98]: done

File: Tools/AP_Periph/Parameters.cpp (hunk 2 above)
Comment (tridge) [reply Tools/AP_Periph/Parameters.cpp:176]: done

**Review by bugobliterator — APPROVED** [review]

**Pull request merged**